src/main/groovy/com/blackducksoftware/integration/hub/detect/util/executable/ExecutableManager.groovy

```diff
@@ -58,56 +62,31 @@ class ExecutableManager {
         }
     }
 
-    String getPathOfExecutable(ExecutableType executableType) {
-        File executableFile = getExecutable(executableType)
-
-        if (executableFile == null) {
-            logger.debug("Expected to find an executable of type ${executableType.toString()} but none was found")
-            return null
-        }
-        return executableFile.absolutePath
-    }
-
-    String getPathOfExecutable(ExecutableType executableType, String defaultPath) {
-        if (defaultPath?.trim()) {
-            if (findExecutableFile(defaultPath.trim())) {
-                return defaultPath.trim()
-            }
-        }
-
-        getPathOfExecutable(executableType)
-    }
-
     String getExecutableName(ExecutableType executableType) {
         executableType.getExecutable(currentOs)
     }
 
-    File getExecutable(ExecutableType executableType) {
-        String executable = executableType.getExecutable(currentOs)
-        File executableFile = findExecutableFile(executable)
-
-        executableFile
-    }
-
-    String getPathOfExecutableFromRelativePath(String path, ExecutableType executableType) {
-        File executableFile = getExecutableFromRelativePath(path, executableType)
-
-        null == executableFile ? null : executableFile.absolutePath
+    String getExecutablePath(ExecutableType executableType, boolean searchSystemPath, String path) {
+        getExecutable(path, executableType, searchSystemPath)?.absolutePath
```

Comment:
Doesnt look like line 70 calls a method that exists

Reply:
Good catch, I forgot to refactor that when I changed the method signature.